File: app/src/main/java/ysports/app/widgets/AutoHideTextView.java
package ysports.app.widgets;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;
import android.util.AttributeSet;
import androidx.annotation.Nullable;

public class AutoHideTextView extends androidx.appcompat.widget.AppCompatTextView {

    private Handler handler;
    private Runnable runnable;

    public AutoHideTextView(Context context) {
        super(context);
    }

    public AutoHideTextView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public AutoHideTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onTextChanged(CharSequence text, int start, int lengthBefore, int lengthAfter) {
        super.onTextChanged(text, start, lengthBefore, lengthAfter);
        if (handler == null) {
            handler = new Handler(Looper.getMainLooper());
        } else {
            handler.removeCallbacksAndMessages(null);
        }
        if (runnable == null) runnable = () -> this.setVisibility(GONE);
        handler.postDelayed(runnable, 1000);
    }
}

File: app/src/main/java/ysports/app/widgets/AutoHideTextView.kt
package ysports.app.widgets

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import ysports.app.R

class AutoHideTextView : AppCompatTextView {

    private var mHandler: Handler? = null
    private var runnable: Runnable? = null
    private var hideMillis: Long = 1000

    constructor(context: Context) : this(context, null)
    constructor(context: Context, attrs: AttributeSet?) : this(
        context, attrs, android.R.attr.textViewStyle
    )

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context, attrs, defStyleAttr
    ) {
        val typedArray = context.theme.obtainStyledAttributes(
            attrs, R.styleable.AutoHideTextView, defStyleAttr, 0
        )
        try {
            hideMillis = typedArray.getInt(R.styleable.AutoHideTextView_hide_millis, 1000).toLong()
        } finally {
            typedArray.recycle()
        }
    }

    override fun onTextChanged(
        text: CharSequence?, start: Int, lengthBefore: Int, lengthAfter: Int
    ) {
        super.onTextChanged(text, start, lengthBefore, lengthAfter)
        if (mHandler == null) mHandler = Handler(Looper.getMainLooper())
        if (runnable == null) runnable = Runnable { this.visibility = GONE }
        mHandler?.removeCallbacks(runnable!!)
        mHandler?.postDelayed(runnable!!, hideMillis)
    }
}